### DISCUSSION TITLE
Is it possible to auto generate docs for Rego based on its metadata?

### DISCUSSION THREAD

**Discussion by renatosc** [open-policy-agent/community#415] [OPA and Rego] [answered]
I looked everywhere for a way to generate documentation from the rego files based on their metadata but I didn't find any. 

Is there an official way to generate such rego doc (html, pdf,...)?

**Comment by anderseknert** [comment, marked as answer]
Hi @renatosc 👋 

There's no official tooling for that currently, although I'd love to see it happen! It's been in my backlog for some time, but unfortunately I haven't had the time to look into it yet. 

If you'd like to try and build something, there's some info in[ the docs](https://www.openpolicyagent.org/docs/latest/policy-language/#go-api) on how to use the Go API to extract annotations from Rego modules, and once you have the data you should be able to format it to your liking (Markdown, HTML, PDF). If you'd like to work on it, I'd be happy to support you. The #contributors channel on the OPA Slack is a good place to start.
